### PR TITLE
Exposing the cookie jar to the options interface

### LINKED ISSUE #23: Cookies are retained across requests
There needs to be a way to disable the request cookie jar. When running selenium tests koa-proxy retains cookies across browser sessions. Even clearing cookies does not stop koa-proxy from forwarding the cookies. Please expose the jar setting to make it easier to disable this behavior.

### PR DESCRIPTION
**Pull request by GreenGremlin**
fixes #23

### CODE REVIEW

**Comment by popomore** [comment]
Can you add a testcase for this option, with jar or not?

**Comment by popomore** [comment]
And make travis happy.

**Comment by GreenGremlin** [comment]
Unit tests added for cookies with and without the cookie jar.

**Comment by popomore** [comment]
Can you use node4 in travis, that's why ci failed.

**Comment by GreenGremlin** [comment]
It looks like the failure was caused by the mz package upgrading their version of any-promise. You'll need to register a promise preference with any-promise before requiring koa-static.

https://github.com/kevinbeaty/any-promise

**Comment by popomore** [comment]
Yes, it's THE Promise adapter, node4 has support it.

File: test/index.js [hunk 1]
```diff
@@ -23,11 +24,21 @@ describe('koa-proxy', function() {
         return;
       }
       if (this.path === '/postme') {
-        this.body = this.req;
+        this.body = this.request.body;
```
Comment (GreenGremlin) [inline test/index.js]: Not sure how this worked before.

**Comment by GreenGremlin** [comment]
@greim, are you able to take a look at the test failures? I've dug in, but I know nothing about streaming. The tests were passing on Node 5.3, but now seem to be failing on 5.6 as well as 4.3.

**Comment by popomore** [comment]
Sorry, I'm on vocation all these days. I will check it in weekend.

**Comment by popomore** [comment]
You can rebase master, it should be fixed by #25

**Comment by popomore** [comment]
And squash to one commit.

**Comment by popomore** [comment]
@GreenGremlin Do you have time?

Or I can pick you code.

**Comment by popomore** [comment]
Land db5278274c52d087e2dee872568c316fb4417353

**Comment by GreenGremlin** [comment]
Sorry, been busy and lost track. Thanks!

**Pull request closed without merging**